protocols/client/client_test.go

```diff
@@ -80,7 +80,7 @@ func TestNewAgentClient(t *testing.T) {
 	assert.Nil(t, err, "failed to start mock server: %s", err)
 
 	cliFunc := func(sock string, success bool, expect string) {
-		cli, err := NewAgentClient(sock)
+		cli, err := NewAgentClient(sock, false)
```

Comment:
Is there no way to unit test the yamux option?

Reply:
surely there is. I'll add UT for it.

Reply:
Thanks @bergwolf ! :smile: